### PR TITLE
Add saml

### PR DESCRIPTION
**Pull request by jparyani**
Not ready to merge as it's missing handling extra fields.

To test:
1. Make a developer account at okta (https://www.okta.com/platform-signup/)
2. Go to the admin panel of okta -> Add Applications -> Create New App
3. Make a saml app. Settings as follows:
Single sign on URL: http://YOUR_HOST/_saml/validate/default
Audience URI: YOUR_HOST (without http://)
Name ID format: EmailAddress (at least until we have other fields working)
4. At the end of setup, click "Setup Instructions"
5. In Sandstorm's admin settings, enable saml and set the entry point as (1) from setup instructions. Set public cert as (3), without the -----BEGIN CERTIFICATE----- parts.

### CODE REVIEW

**Comment by jparyani** [comment]
Now 'displayName' and 'email' are required to be configured as attributes passed to Sandstorm. A not has been added to the admin config page instructing users to do so.

This PR is now ready for review/merge.

File: shell/packages/sandstorm-db/profile.js [hunk 1]
```diff
@@ -269,6 +272,7 @@ SandstormDb.prototype.findIdentitiesByEmail = function (email) {
     { "services.google.email": email },
     { "services.email.email": email },
     { "services.github.emails.email": email },
+    { "services.github.saml.email": email },
```
Comment (kentonv) [inline shell/packages/sandstorm-db/profile.js]: github.saml? :)

File: shell/packages/accounts-saml/saml_server.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ RoutePolicy.declare("/_saml/", "network");
 const HOSTNAME = Url.parse(process.env.ROOT_URL).hostname;
 
 Accounts.registerLoginHandler(function (loginRequest) {
+  if (!Accounts.identityServices.saml.isEnabled()) {
```
Comment (kentonv) [inline shell/packages/accounts-saml/saml_server.js]: Shouldn't this check go *after* the next one (which checks if this is a SAML login)?

File: shell/packages/sandstorm-accounts-ui/login_buttons.html [hunk 3]
```diff
@@ -213,6 +213,12 @@
   </form>
 </template>
 
+<template name="samlLoginForm">
+  <button class="login oneclick saml">
+    with Saml
```
Comment (kentonv) [inline shell/packages/sandstorm-accounts-ui/login_buttons.html]: Saml -> SAML

**Comment by kentonv** [comment]
We should probably extend the About page (found in shell.html) to mention the LDAP and SAML packages.

**Comment by kentonv** [comment]
Garply, retest this please.

**Comment by paulproteus** [comment]
yippee!​

**Pull request merged**